src/model/formulations/base/structs.jl

```diff
@@ -2,28 +2,37 @@
 # Copyright (C) 2020, UChicago Argonne, LLC. All rights reserved.
 # Released under the modified BSD license. See COPYING.md for more details.
 
-abstract type TransmissionFormulation end
-abstract type RampingFormulation end
+abstract type AbstractTransmissionFormulation end
+abstract type AbstractRampingFormulation end
 abstract type PiecewiseLinearCostsFormulation end
-abstract type StartupCostsFormulation end
+abstract type AbstractStartupCostsFormulation end
 abstract type StatusVarsFormulation end
 abstract type ProductionVarsFormulation end
 
+"""
+    struct Formulation
+
+Every formulation has to specify components for setting production variables and limits,
+startup costs and piecewise-linear costs, ramping variables and constraints,
+status variables (on/off, starting up, shutting down), and transmission constraints.
+
+Some of these components are allowed to be empty, as long as overall validity of the formulation is maintained.
+"""
 struct Formulation
     prod_vars::ProductionVarsFormulation
     pwl_costs::PiecewiseLinearCostsFormulation
-    ramping::RampingFormulation
-    startup_costs::StartupCostsFormulation
+    ramping::AbstractRampingFormulation
+    startup_costs::AbstractStartupCostsFormulation
     status_vars::StatusVarsFormulation
-    transmission::TransmissionFormulation
+    transmission::AbstractTransmissionFormulation
```

Comment:
I like the renaming. Not required for this PR, but I think we should also rename `ProductionVarsFormulation`, `PiecewiseLinearCostsFormulation` and `StatusVarsFormulation`.

Reply:
I have reverted this to not use the `Abstract` prefix for this PR, and we can change them all in a later PR.